File: UI/etheater_mobile/lib/models/showSchedule.dart
import 'package:etheater_mobile/models/models.dart';
import 'package:json_annotation/json_annotation.dart';

part 'showSchedule.g.dart';

@JsonSerializable()
class ShowSchedule {
  int showScheduleId;
  int? ticketPrice;
  DateTime showDate;
  String? showTime;
  int? hallId;
  Hall? hall;
  int? showId;
  Show? show;
  bool? isDeleted;

  ShowSchedule(
      {required this.showScheduleId,
      this.ticketPrice,
      required this.showDate,
      this.showTime,
      this.hallId,
      this.hall,
      this.showId,
      this.show,
      this.isDeleted});

  factory ShowSchedule.fromJson(Map<String, dynamic> json) =>
      _$ShowScheduleFromJson(json);
  Map<String, dynamic> toJson() => _$ShowScheduleToJson(this);
}

File: eTheater.Model/ShowSchedule.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace eTheater.Model
{
    public class ShowSchedule
    {
        public int ShowScheduleId { get; set; }

        public int? TicketPrice { get; set; }

        public DateTime? ShowDate { get; set; }

        public string? ShowTime { get; set; }

        public int? ShowId { get; set; }

        public int? HallId { get; set; }

        public virtual Hall? Hall { get; set; }

        public virtual Show? Show { get; set; }

        public bool IsDeleted { get; set; }
    }
}
